wp-parsely-2.5/src/class-parsely.php

```diff
@@ -868,20 +980,26 @@ public static function post_has_trackable_status( $post ) {
 	 * @return mixed|void
 	 */
 	public function construct_parsely_metadata( array $parsely_options, $post ) {
-		$parsely_page = array(
+		$parsely_page      = array(
 			'@context' => 'http://schema.org',
 			'@type'    => 'WebPage',
 		);
-		$current_url  = $this->get_current_url();
+		$current_url       = $this->get_current_url();
+		$queried_object_id = get_queried_object_id();
 
-		if ( is_front_page() && ! is_paged() || ( 'page' === get_option( 'show_on_front' ) && ! get_option( 'page_on_front' ) ) ) {
+		if ( is_front_page() && ! is_paged() ) {
 			$parsely_page['headline'] = $this->get_clean_parsely_page_value( get_bloginfo( 'name', 'raw' ) );
 			$parsely_page['url']      = home_url();
 		} elseif ( is_front_page() && is_paged() ) {
 			$parsely_page['headline'] = $this->get_clean_parsely_page_value( get_bloginfo( 'name', 'raw' ) );
 			$parsely_page['url']      = $current_url;
-		} elseif ( is_home() ) {
-			$parsely_page['headline'] = get_the_title( get_option('page_for_posts', true) );
+		} elseif (
+			is_home() && (
+				! ( 'page' === get_option( 'show_on_front' ) && ! get_option( 'page_on_front' ) ) ||
```

Comment:
`( 'page' === get_option( 'show_on_front' ) && ! get_option( 'page_on_front' )` will never be true? Perhaps a testing artifact?